rexpr.c

```diff
@@ -1585,7 +1585,7 @@ again:;
 		return F->t;
 
 	/* if cant get address, dont */
-	if ((tmpvs [0] = R.tmpv && (!path ? !throuptr : r == 1)))
+	if ((tmpvs [0] = R.tmpv) && (!path ? !throuptr : r == 1))
```

Comment:
Sorry, I duplicate comment due to first time I put it on commit itself, not on PR.

I don't know how the code works. Due to "hackish" code style of original author, this line looks like buggy in old version. And I don't get which way it should work.

But this change changes the way code works. That is why I'm asking: did you really fix some bug with this change, or just put a bracket?

Reply:
I agree with you, I only followed the warnings emitted by gcc and added the bracket.
Somehow lwc now build without warnings and can build all the samples although it seems that they need be revised too, one step at a time.
Again thank you and any feedback is welcome !